### ISSUE TITLE
"false positive" in if_same_then_else when the branches differ by a comment

### ISSUE THREAD

**Issue by nagisa** [rust-lang/rust-clippy#5940]
I tried this code:

```rust
fn foo(a: bool, b: bool) -> Option<bool> {
    if a {
        Some(b)
    } else if b {
        // Some commentary about why this branch is important etc.
        None
    } else if !a {
        // Some more commentary about why this branch matters.
        None
    } else {
        Some(a)
    }
}
```

I expected to see this happen: no false positives!

Instead, this happened: `if_same_then_else` deny-by-default lint fired. While, yes, the branches are indeed "duplicate" at a logical level, they are not at a syntactic level. In this particular case the code ended up this way because it was important to document why we’re returning `None if b` and then `None if !a`, These comments could be rewritten to somehow combine them together, but there's probably nothing that’s clearer than just splitting code into a couple of the `else if` branches seen above. 

### Meta

- `cargo clippy -V`: clippy 0.0.212 (de521cb 2020-08-21)
- `rustc -Vv`:  rustc 1.47.0-nightly (de521cbb3 2020-08-21)

---

(Knowing compilers, this is probably not super actionable, feel free to close if you aren’t too keen on adjusting behaviour here)
